models/marts/core/marketing/dim_customers.sql

```diff
@@ -0,0 +1,99 @@
+with customers as (
+    select * from {{ ref('stg_ecommerce__customers') }}
+    ),
+
+orders as (
+    select * from {{ ref('stg_ecommerce__orders') }}  
+    ),
+
+order_items as (
+    select * from {{ ref('stg_ecommerce__order_items') }} 
+    ),
+
+order_reviews as (
+    select * from {{ ref('stg_ecommerce__order_reviews') }}   
+    ),
+
+cancelled_missing_orders as (
+    select
+    distinct order_id
+    from    orders
+    where  order_status NOT IN ("canceled","unavailable")
+    ),
+
+first_n_last_order as (
+    select
+       customers.customer_id,
+        min(order_approved_at) as first_order_approved_at,
+        max(order_approved_at) as last_order_approved_at,
+        date_diff(max(order_approved_at),min(order_approved_at),DAY) as customer_lifespan
+    from    customers  
```

Comment:
for readability it's best to keep spacing/alignment consistent. Generally we recommend just one space after the `from`